### ISSUE TITLE
symbolic formula changes after forward pass

### ISSUE THREAD

**Issue by CyrillStrassburg** [KindXiaoming/pykan#508]
Hello, first of all, Thank you for your work! I am using KANs to model environmental data and have high hopes.
When training and pruning my KAN i would like to produce metrics and for this i forward pass the test set of my data. While doing this, I noticed that the output of auto_symbolic() changes depending on the last forward pass while everything else stays the same. Why is this? Do forward passes change the model? If so, why? 

Is there a resource i can consult to better understand the code for getting the symbolic formulas?

The behavior is reproducible with the hellokan.ipynb from the repo when changing train and test data to my dataset (and forward passing a torch.rand(10000, 4) ) before calling auto_symbolic. I tried setting the model to evaluation mode and calling torch.no_grad() but it had no effect on the behavior.
